### PR TITLE
PR for statistic endpoint

### PR DESCRIPTION
**Pull request by Josh-121**

### CODE REVIEW

File: web_app/api/user.py [hunk 1]
```diff
@@ -86,3 +88,27 @@ async def get_user_contract_address(wallet_id: str) -> GetUserContractAddressRes
...
+    """
+    try:
+        # Fetch total amounts for each user where position status is OPENED
+        total_amounts = position_db.get_all_amounts_for_opened_positions()
+        
+        # Get the number of unique users
+        unique_users_count = len(total_amounts)
+
+        # Return total amounts and unique users count
+        return {"total_amounts": total_amounts, "unique_users": unique_users_count}
```
Comment (djeck1432) [inline web_app/api/user.py:111]: Add please response mode( create pydantic model) l for this response

File: web_app/api/user.py [hunk 2]
```diff
@@ -86,3 +88,27 @@ async def get_user_contract_address(wallet_id: str) -> GetUserContractAddressRes
...
+    - Dictionary containing:
+        - total_amounts: dict where keys are user IDs and values are total amounts
+        - unique_users: The count of unique users with open positions
+    """
+    try:
+        # Fetch total amounts for each user where position status is OPENED
+        total_amounts = position_db.get_all_amounts_for_opened_positions()
+        
+        # Get the number of unique users
+        unique_users_count = len(total_amounts)
```
Comment (djeck1432) [inline web_app/api/user.py:108]: why do you do `len` for total_amounts, by variable name I understand that this is ready value, not amouns_objects. 
Please rename it and move `len` to `get_all_amounts_for_opened_positions` so then your variable name `total_amounts ` will be correct

File: web_app/api/user.py [hunk 3]
```diff
@@ -86,3 +88,27 @@ async def get_user_contract_address(wallet_id: str) -> GetUserContractAddressRes
...
+
+        # Return total amounts and unique users count
+        return {"total_amounts": total_amounts, "unique_users": unique_users_count}
+
+    except Exception as e:
```
Comment (djeck1432) [inline web_app/api/user.py:113]: which particular exception your are trying to catch?

File: web_app/db/crud.py [hunk 4]
```diff
@@ -326,3 +326,40 @@ def open_position(self, position_id: uuid) -> Position | None:
...
+    
+    def get_unique_users(self) -> list[str]:
+        """
+        Fetches all unique users from the Position table.
+        :return: list of unique user ids
+        """
+        with self.Session() as db :
+            try:
+                # Query for distinct user IDs from the Position table
+                unique_users = db.query(Position.user_id).distinct().all()
```
Comment (djeck1432) [inline web_app/db/crud.py:338]: please, use `.count()` method for calculating amount of unique users

File: web_app/db/crud.py [hunk 5]
```diff
@@ -326,3 +326,40 @@ def open_position(self, position_id: uuid) -> Position | None:
...
+        
+        :return: A dictionary with user IDs as keys and the total amount as values.
+        """
+        with self.Session() as db:
+            try:
+                # Query all positions with status OPENED and group by user_id
+                opened_positions = (
+                    db.query(Position.user_id, db.func.sum(Position.amount).label('total_amount'))
+                    .filter(Position.status == Status.OPENED.value)
+                    .group_by(Position.user_id)
```
Comment (djeck1432) [inline web_app/db/crud.py:356]: Why do you need to group by `users`?
```
total_opened_amount = (
    db.query(db.func.sum(Position.amount))
    .filter(Position.status == Status.OPENED.value)
    .scalar()
)
```

File: web_app/db/crud.py [hunk 6]
```diff
@@ -326,3 +326,40 @@ def open_position(self, position_id: uuid) -> Position | None:
...
+                    .all()
+                )
+                
+                # Return a dictionary with user IDs and their respective total amounts
+                return {user_id: total_amount for user_id, total_amount in opened_positions}
+
+            except SQLAlchemyError as e:
+                logger.error(f"Error fetching amounts for opened positions: {str(e)}")
+                return {}
```
Comment (djeck1432) [inline web_app/db/crud.py:365]: run `black` command

**Review by djeck1432 — CHANGES_REQUESTED** [review]

File: web_app/db/crud.py (hunk 4 above)
Comment (Josh-121) [reply web_app/db/crud.py:343]: i'm returning a list containing all unique user id's. Am I to return just the total count?

File: web_app/db/crud.py (hunk 5 above)
Comment (Josh-121) [reply web_app/db/crud.py:361]: here i'm returning a dic of user ids & their corresponding amount for opened positions.  I'll rename the variables for better clarity
or am I to return something different from this?

File: web_app/api/serializers/user.py [hunk 7]
```diff
@@ -27,8 +29,24 @@ class GetUserContractAddressResponse(BaseModel):
...
     """
+
     contract_address: str | None = Field(
         None,
         example="0xabc123...",
         description="The contract address of the user, or None if not deployed.",
     )
+
+
+class GetStats(BaseModel):
```
Comment (djeck1432) [inline web_app/api/serializers/user.py:40]: Model name can start with `verb` 
Change it to `StatsData` or `StatsResponse` if it's response model

File: web_app/api/user.py (hunk 3 above)
Comment (djeck1432) [reply web_app/api/user.py:162]: still opened issue

File: web_app/db/crud.py (hunk 4 above)
Comment (djeck1432) [reply web_app/db/crud.py:342]: > Am I to return just the total count?
Yes, it's

File: web_app/db/crud.py (hunk 5 above)
Comment (djeck1432) [reply web_app/db/crud.py:363]: Yeah, we need to have result as a amount of all `amount` for `opened` position
Just copy/paste what I made as an example above

**Review by djeck1432 — CHANGES_REQUESTED** [review]

**Review by djeck1432 — APPROVED** [review]
Looks good to me

**Pull request merged**